AUTHORING_GUIDE.md

```diff
@@ -281,216 +191,203 @@ Args:
+Sample code may be integrated into Google Cloud Documentation through the use
+of region tags, which are comments added to the source code to identify code
+blocks that correspond to specific topics covered in the documentation. For
+example, see
+[this sample](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/cloud-sql/mysql/sqlalchemy/main.py)
+— the region tags are the comments that begin with `[START` or `[END`.
 
-For command-line samples, you'll need an argparse section to handle
-parsing command-line arguments and executing the sample functions. This
-section lives within the `if __name__ == '__main__'` clause:
+The use of region tags is beyond the scope of this document, but if you’re
+using region tags they should start after the source code header
+(license/copyright information), imports, and global configuration such as
+initializing constants.
 
-```python
-if __name__ == '__main__':
-    parser = argparse.ArgumentParser(
-        description=__doc__,
-        formatter_class=argparse.RawDescriptionHelpFormatter)
-```
+## Exception Handling
 
-Note the use of `__doc__` (the module-level docstring) as the description for
-the parser. This helps you not repeat yourself and gives users useful
-information when invoking the program. We also use `RawDescriptionHelpFormatter`
-to prevent argparse from re-formatting the docstring.
+Sample code should use standard Python exception handling techniques as
+covered in the [Google Python Style
+Guide](http://google.github.io/styleguide/pyguide.html#24-exceptions).
 
-Command-line arguments should generally have a 1-to-1 match to function
-arguments. For example:
+# Testing Guidelines
 
-```python
-parser.add_argument('source_file_name')
-parser.add_argument('destination_blob_name')
-```
+Samples should include tests to verify that the sample runs correctly and
+generates the intended output. Follow these guidelines while writing your
+tests:
+
+* Use [pytest](https://docs.pytest.org/en/latest/)-style tests and plain
+asserts. Don't use `unittest`-style tests or `assertX` methods.
+* Whenever possible, tests should avoid taking a dependency on the precise
+output of APIs. For example, if a test is intended to verify a JSON payload
+returned from an endpoint, it should only check for the existence of the
+expected keys and values, and the test should continue to work correctly
+if other values are added to the response in a future version of the API.
+* Samples that use App Engine Standard should use the [App Engine
+testbed](https://cloud.google.com/appengine/docs/standard/python/refdocs/google.appengine.ext.testbed)
+for system testing, as shown in [this
+example](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/appengine/standard/localtesting/datastore_test.py).
+* All tests should be independent of one another and order-independent.
 
-Again, descriptive names prevent you from having to exhaustively describe
-every parameter.
+## Arrange, Act, Assert
```

Comment:
This is actually a rule of thumb for the snippets themselves. Most snippets should use the follow pattern:

1. Arrange - construct the API request and build the different components needed to configure the call
2. Act - send the request to the server and wait for a response
3. Assert - verify the response is a success, and act upon it (usually printing some results to show the call was successful)